logstash-core/lib/logstash/java_pipeline.rb

```diff
@@ -238,7 +238,7 @@ def start_workers
           Util.set_thread_name("[#{pipeline_id}]>worker#{t}")
           org.logstash.execution.WorkerLoop.new(
               lir_execution, filter_queue_client, @events_filtered, @events_consumed,
-              @flushRequested, @flushing, @shutdownRequested, @drain_queue).run
```

Comment:
nitpick, and potentially orthogonal to this fix:

I see value in ordering guarantees being made explicit opt-in at some point (or at least to make room for an explicit opt-out), since in many cases it doesn't matter and making ordering guarantees incurs overhead.

For this I propose a new pipeline setting `pipeline.ordered`; in 7.x, this should default to true IFF `pipeline.workers==1` to preserve the observed behaviour of the Ruby engine, but in 8.x the setting would need to be explicitly specified to incur the overhead to get the guarantee.

|  | v7.x | v8.x |
|--|--|--|
| default | `pipeline.workers == 1` | `false` |

Reply:
That occurred to me, too. No need to incur the overhead if event ordering is unimportant, but definitely something that should be done in a separate PR.